sdk/tools/com-archive-tool/src/main/java/esa/mo/nmf/com_archive_browser/ArchiveBrowserHelper.java

```diff
@@ -0,0 +1,329 @@
+    @Option(names = {"-l", "--local"}, paramLabel = "<databaseFile>",
+            description = "Local SQLite database file\n"
+                          + "  - example: ../nanosat-mo-supervisor-sim/comArchive.db")
+    public String databaseFile;
+
+    @Option(names = {"-r", "--remote"}, paramLabel = "<providerURI>",
+            description = "Remote COM archive provider URI\n"
+                          + "  - example: maltcp://10.0.2.15:1024/nanosat-mo-supervisor-Archive")
+    public String providerURI;
+  }
+
+  /**
+   * Search a COM archive provider content to find the ObjectId of an App of the CommandExecutor
+   * service of the SoftwareManagement.
+   *
+   * @param appName Name of the NMF app we want the logs for
+   * @param domain Restricts the search to objects in a specific domain ID
+   * @return the ObjectId of the found App or null if not found
+   */
+  public static ObjectId getAppObjectId(String appName,
+      IdentifierList domain, ArchiveConsumerServiceImpl consumer) {
+    // SoftwareManagement.AppsLaunch.App object type
+    ObjectType appType = new ObjectType(SoftwareManagementHelper.SOFTWAREMANAGEMENT_AREA_NUMBER,
+        AppsLauncherHelper.APPSLAUNCHER_SERVICE_NUMBER, new UOctet((short) 0),
+        AppsLauncherHelper.APP_OBJECT_NUMBER);
+
+    // prepare domain filter
+    ArchiveQueryList archiveQueryList = new ArchiveQueryList();
+    ArchiveQuery archiveQuery =
+        new ArchiveQuery(domain, null, null, 0L, null, null, null, null, null);
+    archiveQueryList.add(archiveQuery);
+
+    // execute query
+    ArchiveToAppAdapter adapter = new ArchiveToAppAdapter(appName);
+    queryArchive(appType, archiveQueryList, adapter, adapter, consumer);
+    return adapter.getAppObjectId();
+  }
+
+  /**
+   * Queries objects from a COM archive provider.
+   *
+   * @param objectsTypes COM types of objects to query
+   * @param archiveQueryList Archive query object used for filtering
+   * @param adapter Archive adapter receiving the query answer messages
+   * @param queryStatusProvider Interface providing the status of the query
+   */
+  public static void queryArchive(ObjectType objectsTypes,
+    ArchiveQueryList archiveQueryList, ArchiveAdapter adapter,
+    QueryStatusProvider queryStatusProvider, ArchiveConsumerServiceImpl consumer) {
+    // run the query
+    try {
+      consumer.getArchiveStub().query(true, objectsTypes, archiveQueryList, null, adapter);
+    } catch (MALInteractionException | MALException e) {
+      LOGGER.log(Level.SEVERE, "Error when querying archive", e);
+      return;
+    }
+
+    // wait for query to end
+    while (!queryStatusProvider.isQueryOver()) {
+      try {
+        Thread.sleep(500);
+      } catch (InterruptedException ignored) {
+      }
+    }
+  }
+
+  public static void closeLocalProvider() {
+    if(localProvider != null) {
+      localProvider.close();
+    }
+  }
+
+  public static LocalOrRemoteConsumer createConsumer(String providerURI, String databaseFile) {
```

Comment:
Could you please make sure that at least public methods have a minimal Java doc? I know it's annoying to do but it greatly helps when we come back to the code months later :)